roles/openshift_health_checker/openshift_checks/sdn.py

```diff
@@ -0,0 +1,698 @@
+            # Skip the pod if it is not on this node.
+            if pod['spec'].get('nodeName') != node_name:
+                continue
+
+            # Skip the pod if it does not have a ready container or if it is
+            # on the host's network.
+            status = pod['status']
+            if 'containerStatuses' not in status or \
+               status['containerStatuses'][0].get('ready') is not True or \
+               'hostNetwork' in pod['spec']:
+                continue
+
+            # Get the pid of a container in the pod; we will need the pid in
+            # order to enter the container's netns.
+            cid = status['containerStatuses'][0]['containerID'] \
+                .replace('docker://', '')
+            try:
+                pid = self.read_command_output(['docker', 'inspect',
+                                                '-f', '{{.State.Pid}}', cid])
+                if not pid or pid == '0':
+                    raise OpenShiftCheckException(
+                        'CouldNotReadPodPid',
+                        'Could not determine pid of pod %s in namespace %s' %
+                        (pod['metadata']['name'],
+                         pod['metadata']['namespace']))
+                pod['pid'] = pid
+            except OpenShiftCheckException as exc:
+                self.register_failure(exc)
+                continue
+
+            # Look up and store the MAC address, OVS bridge port and VNID for
+            # the pod so that the connectivity checks can easily look them up
+            # later.
+            port = pod_addr_to_port.get(status['podIP'], None)
+            if not port:
+                self.register_failure('Could not find port for %s!' %
+                                      status['podIP'])
+                continue
+            pod['ovs_bridge_port'] = port
+
+            vnid = pod_addr_to_vnid.get(status['podIP'], '0')
+            if not vnid:
+                self.register_failure('Could not find VNID for %s!' %
+                                      status['podIP'])
+                continue
+            pod['vnid'] = vnid
+
+            # XXX: Would it be worthwhile to create an nsenter module?
+            nsenter_argv = ['nsenter', '-n', '-t', pod['pid'], '--']
+
+            addresses = self.read_command_output(nsenter_argv +
+                                                 ['ip', 'addr', 'show'])
+
+            self.register_file('pod_%s_addresses' % pod['metadata']['name'],
+                               addresses)
+            self.save_command_output('pod_%s_routes' % pod['metadata']['name'],
+                                     nsenter_argv + ['ip', 'route', 'show'])
+
+            tokens = addresses.split()
+            idx = tokens.index('link/ether')
+            if idx and idx + 1 < len(tokens):
+                pod['mac_addr'] = tokens[idx + 1]
+                # XXX: Should we check that there is only one Ethernet
```

Comment:
It would be nice to log something if you got more than one.  It would be surprising.

Reply:
Changed to register a failure if there is not exactly 1 MAC address.